server/src/main/java/org/opensearch/tasks/TaskInfo.java

```diff
@@ -64,6 +65,9 @@
  * snapshot information about currently running tasks.
  */
 public final class TaskInfo implements Writeable, ToXContentFragment {
+
+    static final String INCLUDE_CANCELLED_PARAM = "include_cancelled";
```

Comment:
I am not super familiar with this code, why is this necessary?

Reply:
We don’t want this field to be shown if task is completed and also as an entry in the task index as we are not mapping this dynamically.

Reply:
Where else are we doing something similar?

Reply:
I also don't see any new tests that utilize this behavior.

Reply:
Why don't we want this field to be shown if the task is completed?

If being completed it intended to be mutually exclusive from being cancelled, did you consider adding something like a `state` field where the value could be one of `RUNNING`, `CANCELLED` or `COMPLETED`?

Reply:
As the cancelled field tells about the task being cancelled or not, having it when the task is completed felt unnecessary. If the task is completed it means that it has not been cancelled and as mentioned in the issue, if the task is cancelled it will eventually abort/fail. I think the cancelled field does the job of the state field that you mentioned as it shows the `false` = task is running and `true` = task is cancelled.

Reply:
> it shows the false = task is running and true = task is cancelled

The logic here implicitly has a third state of the cancelled boolean, which is "not present = task is complete". If there is a strong reason to have that third state then using a type that can explicitly represent three states is a better way to go. However, I'd consider doing the simpler approach of always including the cancelled boolean, even on complete tasks when it will be set to false.

Reply:
I have added the cancelled field to be shown as a field when the task is completed.